src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt

```diff
@@ -96,15 +100,50 @@ class LicenceService(
     }
   }
 
+  @Transactional
+  fun updateLicenceStatus(licenceId: Long, request: StatusUpdateRequest) {
+    val licenceEntity = licenceRepository
+      .findById(licenceId)
+      .orElseThrow { EntityNotFoundException("$licenceId") }
+
+    val updatedLicence = licenceEntity.copy(
+      statusCode = request.status,
+      dateLastUpdated = LocalDateTime.now(),
+      updatedByUsername = request.username,
+      approvedByUsername = request.username.takeIf { request.status == APPROVED } ?: licenceEntity.approvedByUsername,
```

Comment:
If the licence is already approved, and we use this endpoint to set the status back to IN_PROGRESS, shouldnt the `approvedBy` fields get blanked out, since it requires approval again?

Reply:
Yes, they should! I'll add this too, thanks. I'm sure there will be others when we think of them.. need a state-transition diagram..